File: 백준/Bronze/5566. 주사위 게임/주사위 게임.py
import sys

# 입력 받기
n, m = map(int, input().split())
board = [int(input()) for _ in range(n)]
dice = [int(input()) for _ in range(m)]

# 초기 위치와 결과 값 설정
pos = 1
result = 0

# 주사위 굴리기
for num in dice:
    pos += num
    result += 1
    if pos >= n:
        break
    instruction = board[pos - 1]
    pos += instruction
    if pos >= n:
        break

print(result)

File: 백준/Bronze/5566. 주사위 게임/주사위 게임.java
import java.io.*;
import java.util.ArrayList;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        String str = bufferedReader.readLine();
        StringTokenizer st = new StringTokenizer(str);
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        ArrayList<Integer> board = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            str = bufferedReader.readLine();
            board.add(Integer.parseInt(str));
        }

        ArrayList<Integer> dice = new ArrayList<>();
        for (int i = 0; i < m; i++) {
            str = bufferedReader.readLine();
            dice.add(Integer.parseInt(str));
        }

        int pos = 1;
        int result = 0;
        for (Integer num : dice) {
            pos += num;
            result++;
            if (pos >= n) {
                break;
            }
            Integer instruction = board.get(pos - 1);
            pos += instruction;
            if (pos >= n) {
                break;
            }
        }

        System.out.println(result);
    }
}